src/CoreFoundation/CFException.cs

```diff
@@ -115,7 +115,7 @@ public static CFException FromCFError (IntPtr cfErrorHandle, bool release)
 			if (cfUserInfo != IntPtr.Zero) {
 				using (var userInfo = new NSDictionary (cfUserInfo)) {
 					foreach (var i in userInfo)
-						e.Data.Add (i.Key, i.Value);
+						e.Data.Add (i.Key?.ToString (), i.Value?.ToString ());
```

Comment:
I couldn't find any API that convert instances of NSObject to a corresponding managed instance, so I just convert dictionary keys/values  to strings.

We could add API that looks at the NSObjects and figure out what managed types those correspond to (NSString -> string, NSNumber -> corresponding number, etc), but it felt a bit too complex and open-ended.